lock-impl/src/test/java/com/palantir/lock/LockServiceTest.java

```diff
@@ -840,47 +833,39 @@ public Void call() throws InterruptedException {
                                             SimpleTimeDuration.of(2000, TimeUnit.MILLISECONDS))
                                     .build();
                         }
+
                         token = server.lockWithFullLockResponse(client, request).getToken();
+
                         if (token == null) {
-                            numFailure.set(numFailure.get() + 1);
-                        } else {
-                            numSuccess.set(numSuccess.get() + 1);
-                            Assert.assertEquals(Integer.toString(clientID), token.getClient().getClientId());
-                            Assert.assertEquals(request.getLockDescriptors(), token.getLockDescriptors());
-                            try {
-                                Thread.sleep(50);
-                            } catch (InterruptedException e) {
-                                /* Intentionally swallow. */
-                            }
-                            System.out.println(System.currentTimeMillis()
-                                    - token.getExpirationDateMs());
+                            continue;
+                        }
+
+                        Assert.assertEquals(Integer.toString(clientID), token.getClient().getClientId());
+                        Assert.assertEquals(request.getLockDescriptors(), token.getLockDescriptors());
+                        try {
                             server.unlock(token);
-                            Assert.assertTrue(server.getTokens(client).isEmpty());
+                        } catch (Exception e) {
+                            Assert.assertTrue(e instanceof InterruptedException);
+                            // make sure we unlock before terminating
+                            // each future is only cancelled once, so no need for general retries here
+                            server.unlock(token);
+                            throw e;
                         }
-                    }
-                    } catch (RuntimeException e) {
-                        e.printStackTrace();
-                        throw e;
+                        Assert.assertTrue(server.getTokens(client).isEmpty());
                     }
                 }
             };
+
             futures.add(future);
             executor.execute(future);
         }
+
         Thread.sleep(5000);
+
         for (Future<?> future : futures) {
             future.cancel(true);
         }
-        for (Future<?> future : futures) {
```

Comment:
I think you need this block -- otherwise the test won't wait for the futures to finish before it ends, which is weird.